src/Symfony/Component/HttpFoundation/HeaderBag.php

```diff
@@ -121,7 +121,7 @@ public function get($key, $default = null, $first = true)
         }
 
         if ($first) {
-            return \count($headers[$key]) ? $headers[$key][0] : $default;
+            return \count($headers[$key]) ? (string) $headers[$key][0] : $default;
```

Comment:
I think this introduces a regression for the `Expires` header. It seems to be that this causes the `Expires` header to be written too often, because the check in the following line will now always be true, because `$value` will now be an empty string if it is not set:

https://github.com/symfony/symfony/blob/0890970/src/Symfony/Component/HttpFoundation/HeaderBag.php#L208

Reply:
@nicolas-grekas I would be willing to contribute a fix. Should we just check for empty string instead of null?

Reply:
I think the fix should be ok this line: we should allow null to be returned.

Reply:
Okay, I'm on it, and will add a test for the expires header.

Reply:
Created the PR: https://github.com/symfony/symfony/pull/33353